Marlin/src/module/planner.cpp

```diff
@@ -3053,6 +3057,19 @@ bool Planner::buffer_line(const xyze_pos_t &cart, const_feedRate_t fr_mm_s
 
   #else // !IS_KINEMATIC
 
+    // Differential Extruder
+    // The sum of X and E movement is applied to the E stepper.
+    // Outside of a print job, the differential extruder is handled by stepper.cpp.
+    #if ENABLED(DIFFERENTIAL_EXTRUDER)
+      if (print_job_timer.isRunning()) {
+        // Calculate the differential steps for the extruder
+        const int32_t e_steps = lround(machine.e * steps_per_mm[E_AXIS])
+                              + lround(machine.x * steps_per_mm[X_AXIS]);
```

Comment:
We provide `steps_per_mm` specifically to avoid division in converting mm to steps.

Reply:
Thanks. That looks better. FYI, I edited most of the code with a lot of help from CoPilot. C++ novice...

Reply:
I was getting an error that steps_per_mm was not defined, and Copilot's suggestion was to use settings.axis_steps_per_mm instead. The end result for the statement in lines 3066 and 3067 was:

const int32_t e_steps = lround(machine.e * settings.axis_steps_per_mm[E_AXIS])
                              + lround(machine.x * settings.axis_steps_per_mm[X_AXIS]);

The test print came out good